[BUG] Jaxrs-Jersey produces server that doesn't compile + doesn't handle parent tag in pom.xml correctly

##### Description

I've been testing out `openapi-generator` but have come up against a few issues, which I've repro'd using the petstore example found at  https://github.com/OAI/OpenAPI-Specification/blob/master/examples/v3.0/petstore.yaml. I've been trying out server generation by running `openapi-generator generate -i petstore.yaml -g jaxrs-jersey -o server`, but have encountered the following issues:
1. After generation, when I run `mvn clean package jetty:run` as instructed by the generated README, I get the following compilation error:
```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] ~/repos/OpenAPI-Specification/examples/v3.0/server/src/gen/java/org/openapitools/api/ApiResponseMessage.java:[3,33] package javax.xml.bind.annotation does not exist
[ERROR] ~/repos/OpenAPI-Specification/examples/v3.0/server/src/gen/java/org/openapitools/api/ApiResponseMessage.java:[5,27] package javax.xml.bind.annotation does not exist
[ERROR] ~/repos/OpenAPI-Specification/examples/v3.0/server/src/gen/java/org/openapitools/api/ApiResponseMessage.java:[45,6] cannot find symbol
  symbol:   class XmlTransient
  location: class org.openapitools.api.ApiResponseMessage
```
2. If I run the generation step with `openapi-generator generate -i petstore.yaml -g jaxrs-jersey -o server --additional-properties=parentArtifactId=parent-artifact,parentGroupId=parent-group,parentVersion=1.0-SNAPSHOT`, then I don't see any `parent` tag in the generated `pom.xml` (as expected after reading the docs at https://openapi-generator.tech/docs/generators/jaxrs-jersey/). This is also true with the `jaxrs-resteasy` generator (where, again, the docs at https://openapi-generator.tech/docs/generators/jaxrs-resteasy/ say it's an available option).

##### openapi-generator version
```
> openapi-generator version
5.0.1
```

##### OpenAPI declaration file content or url

Repro'd with the petstore example:
https://github.com/OAI/OpenAPI-Specification/blob/master/examples/v3.0/petstore.yaml

##### Generation Details

`openapi-generator generate -i petstore.yaml -g jaxrs-jersey -o server` & `openapi-generator generate -i petstore.yaml -g jaxrs-jersey -o server --additional-properties=parentArtifactId=parent-artifact,parentGroupId=parent-group,parentVersion=1.0-SNAPSHOT`

##### Steps to reproduce

Run the above generation commands from inside https://github.com/OAI/OpenAPI-Specification/tree/master/examples/v3.0.

##### Related issues/PRs

n/a

##### Suggest a fix

Unsure

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/cxf-cdi/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/cxf-cdi/pom.mustache
@@ -6,6 +6,14 @@
     <name>{{artifactId}}</name>
     <version>{{artifactVersion}}</version>
 
+{{#parentOverridden}}
+    <parent>
+        <groupId>{{parentGroupId}}</groupId>
+        <artifactId>{{parentArtifactId}}</artifactId>
+        <version>{{parentVersion}}</version>
+    </parent>
+{{/parentOverridden}}
+
     <build>
         <!-- Main source directory -->
         <sourceDirectory>src/main/java</sourceDirectory>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/cxf-ext/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/cxf-ext/pom.mustache
@@ -8,6 +8,15 @@
   <description>{{appDescription}}</description>
   {{/appDescription}}
   <version>{{artifactVersion}}</version>
+
+{{#parentOverridden}}
+  <parent>
+    <groupId>{{parentGroupId}}</groupId>
+    <artifactId>{{parentArtifactId}}</artifactId>
+    <version>{{parentVersion}}</version>
+  </parent>
+{{/parentOverridden}}
+
   <build>
     <sourceDirectory>src/main/java</sourceDirectory>
     <plugins>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/cxf-ext/server/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/cxf-ext/server/pom.mustache
@@ -15,6 +15,15 @@
     <version>2.0.4.RELEASE</version>
   </parent>
 {{/generateSpringBootApplication}}
+{{^generateSpringBootApplication}}
+{{#parentOverridden}}
+  <parent>
+    <groupId>{{parentGroupId}}</groupId>
+    <artifactId>{{parentArtifactId}}</artifactId>
+    <version>{{parentVersion}}</version>
+  </parent>
+{{/parentOverridden}}
+{{/generateSpringBootApplication}}
   <build>
     <sourceDirectory>src/main/java</sourceDirectory>
     <plugins>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/cxf/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/cxf/pom.mustache
@@ -8,6 +8,15 @@
   <description>{{appDescription}}</description>
   {{/appDescription}}
   <version>{{artifactVersion}}</version>
+
+{{#parentOverridden}}
+  <parent>
+    <groupId>{{parentGroupId}}</groupId>
+    <artifactId>{{parentArtifactId}}</artifactId>
+    <version>{{parentVersion}}</version>
+  </parent>
+{{/parentOverridden}}
+
   <build>
     <sourceDirectory>src/main/java</sourceDirectory>
     <plugins>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/cxf/server/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/cxf/server/pom.mustache
@@ -8,6 +8,15 @@
   <description>{{appDescription}}</description>
   {{/appDescription}}
   <version>{{artifactVersion}}</version>
+
+{{#parentOverridden}}
+  <parent>
+    <groupId>{{parentGroupId}}</groupId>
+    <artifactId>{{parentArtifactId}}</artifactId>
+    <version>{{parentVersion}}</version>
+  </parent>
+{{/parentOverridden}}
+
   <build>
     <sourceDirectory>src/main/java</sourceDirectory>
     <plugins>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/libraries/jersey1/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/libraries/jersey1/pom.mustache
@@ -5,6 +5,15 @@
   <packaging>jar</packaging>
   <name>{{artifactId}}</name>
   <version>{{artifactVersion}}</version>
+
+{{#parentOverridden}}
+  <parent>
+    <groupId>{{parentGroupId}}</groupId>
+    <artifactId>{{parentArtifactId}}</artifactId>
+    <version>{{parentVersion}}</version>
+  </parent>
+{{/parentOverridden}}
+
   <build>
     <sourceDirectory>src/main/java</sourceDirectory>
     <plugins>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/pom.mustache
@@ -6,6 +6,14 @@
   <name>{{artifactId}}</name>
   <version>{{artifactVersion}}</version>
 
+{{#parentOverridden}}
+  <parent>
+    <groupId>{{parentGroupId}}</groupId>
+    <artifactId>{{parentArtifactId}}</artifactId>
+    <version>{{parentVersion}}</version>
+  </parent>
+{{/parentOverridden}}
+
   <licenses>
     <license>
       <name>{{licenseName}}</name>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/resteasy/eap/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/resteasy/eap/pom.mustache
@@ -6,6 +6,15 @@
     <packaging>war</packaging>
     <name>{{artifactId}}</name>
     <version>{{artifactVersion}}</version>
+
+{{#parentOverridden}}
+    <parent>
+        <groupId>{{parentGroupId}}</groupId>
+        <artifactId>{{parentArtifactId}}</artifactId>
+        <version>{{parentVersion}}</version>
+    </parent>
+{{/parentOverridden}}
+
     <build>
         <sourceDirectory>src/main/java</sourceDirectory>
         <plugins>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/resteasy/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/resteasy/pom.mustache
@@ -6,6 +6,14 @@
     <packaging>war</packaging>
     <name>{{artifactId}}</name>
     <version>{{artifactVersion}}</version>
+
+{{#parentOverridden}}
+    <parent>
+        <groupId>{{parentGroupId}}</groupId>
+        <artifactId>{{parentArtifactId}}</artifactId>
+        <version>{{parentVersion}}</version>
+    </parent>
+{{/parentOverridden}}
     <build>
         <sourceDirectory>src/main/java</sourceDirectory>
         <plugins>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/spec/libraries/helidon/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/spec/libraries/helidon/pom.mustache
@@ -7,6 +7,14 @@
     <artifactId>{{artifactId}}</artifactId>
     <version>{{artifactVersion}}</version>
 
+{{#parentOverridden}}
+    <parent>
+        <groupId>{{parentGroupId}}</groupId>
+        <artifactId>{{parentArtifactId}}</artifactId>
+        <version>{{parentVersion}}</version>
+    </parent>
+{{/parentOverridden}}
+
     <properties>
         <mainClass>io.helidon.microprofile.server.Main</mainClass>
         <version.lib.helidon>1.2.0</version.lib.helidon>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/spec/libraries/kumuluzee/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/spec/libraries/kumuluzee/pom.mustache
@@ -8,6 +8,14 @@
     <artifactId>{{artifactId}}</artifactId>
     <version>{{artifactVersion}}</version>
 
+{{#parentOverridden}}
+    <parent>
+        <groupId>{{parentGroupId}}</groupId>
+        <artifactId>{{parentArtifactId}}</artifactId>
+        <version>{{parentVersion}}</version>
+    </parent>
+{{/parentOverridden}}
+
     <name>Automatically generated KumuluzEE microservice</name>
     <description>The microservice was generated automatically with the OpenAPI Generator project</description>
 

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/spec/libraries/openliberty/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/spec/libraries/openliberty/pom.mustache
@@ -3,11 +3,20 @@
     xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
+{{#parentOverridden}}
+    <parent>
+        <groupId>{{parentGroupId}}</groupId>
+        <artifactId>{{parentArtifactId}}</artifactId>
+        <version>{{parentVersion}}</version>
+    </parent>
+{{/parentOverridden}}
+{{^parentOverridden}}
     <parent>
         <groupId>net.wasdev.wlp.maven.parent</groupId>
         <artifactId>liberty-maven-app-parent</artifactId>
         <version>2.6</version>
     </parent>
+{{/parentOverridden}}
 
     <groupId>{{groupId}}</groupId>
     <artifactId>{{artifactId}}</artifactId>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/spec/libraries/quarkus/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/spec/libraries/quarkus/pom.mustache
@@ -6,6 +6,15 @@
   <artifactId>{{artifactId}}</artifactId>
   <name>{{artifactId}}</name>
   <version>{{artifactVersion}}</version>
+
+{{#parentOverridden}}
+  <parent>
+    <groupId>{{parentGroupId}}</groupId>
+    <artifactId>{{parentArtifactId}}</artifactId>
+    <version>{{parentVersion}}</version>
+  </parent>
+{{/parentOverridden}}
+
   <properties>
     <compiler-plugin.version>3.8.1</compiler-plugin.version>
     <maven.compiler.parameters>true</maven.compiler.parameters>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/spec/libraries/thorntail/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/spec/libraries/thorntail/pom.mustache
@@ -7,6 +7,15 @@
   <name>{{artifactId}}</name>
   <version>{{artifactVersion}}</version>
   <packaging>war</packaging>
+
+{{#parentOverridden}}
+  <parent>
+    <groupId>{{parentGroupId}}</groupId>
+    <artifactId>{{parentArtifactId}}</artifactId>
+    <version>{{parentVersion}}</version>
+  </parent>
+{{/parentOverridden}}
+
   <properties>
     <version.thorntail>2.5.0.Final</version.thorntail>
     <version.junit>4.8.1</version.junit>

--- a/modules/openapi-generator/src/main/resources/JavaJaxRS/spec/pom.mustache
+++ b/modules/openapi-generator/src/main/resources/JavaJaxRS/spec/pom.mustache
@@ -5,6 +5,15 @@
   <packaging>{{#interfaceOnly}}jar{{/interfaceOnly}}{{^interfaceOnly}}war{{/interfaceOnly}}</packaging>
   <name>{{artifactId}}</name>
   <version>{{artifactVersion}}</version>
+
+{{#parentOverridden}}
+  <parent>
+    <groupId>{{parentGroupId}}</groupId>
+    <artifactId>{{parentArtifactId}}</artifactId>
+    <version>{{parentVersion}}</version>
+  </parent>
+{{/parentOverridden}}
+
   <build>
     <sourceDirectory>src/main/java</sourceDirectory>
     <plugins>

